### PR TITLE
Creating an idealised example

### LINKED ISSUE #51: examples
Hi all,

Would it be possible to include at least ~1-2 examples where all the input datasets are publicly available? 

More precisely, after installing `plasticparcels`, I can't reproduce any of the examples, as e.g. [example_Italy_coast](https://plastic.oceanparcels.org/en/latest/examples/example_Italy_coast.html)  requires the model settings:

```
# Load the model settings
settings_file = 'docs/examples/example_Italy_coast_settings.json'
settings = pp.utils.load_settings(settings_file)
```
which are not part of the path. 

Also, looking at those example configuration files [in the repo](https://github.com/OceanParcels/plasticparcels/tree/main/docs/examples), they all rely on private storage `/storage/shared/oceanparcels/input_data/`.

openjournals/joss-reviews#7094

### PR DESCRIPTION
**Pull request by michaeldenes**
This pull-request is to include a new example notebook, using an idealised flow (the Bickley Jet) to show how `plasticparcels` can be used, without the need to download any datasets.

With #56 , this is to close #51

### CODE REVIEW

**Comment by erikvansebille** [comment]
I can't comment on the ipynb file using GitHub tools, so here a list of (minor) comments:
- [ ] In cell 4, why not simply skip the VectorField plots (and use a counter that increments after every valid Field has been plotted)?
- [ ] Before cell 5, perhaps provide links to other examples where the settings dictionary _can_ be loaded?
- [ ] In cell 5, mention where the b/c constants come from? Some values seem arbitrary, otherwise?
- [ ] Perhaps for a later PR, but could the setting of fieldset-constants from the settings-dict not be a method?
- [ ] In cell 10 (and later), how would the user know it's `kernels[-2]` that needs replacing?
- [ ] In cell 29, perhaps print the final depth range with fewer digits?

File: docs/examples/idealised_flow.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    The temperature/salinity field is time-invariant and has a linear decay with depth.
+    Of course, this makes no sense in reality, but it is a simple example."""
+    lon = fieldset.U.grid.lon #TODO Given the Bickley Jet above is C-Grid, is this the right approach to create the other velocity fields??
+    lat = fieldset.U.grid.lat
+    depth = fieldset.U.grid.depth
+
+    T = np.zeros((times.size, depth.size, lat.size, lon.size), dtype=np.float32)
+    S = np.zeros((times.size, depth.size, lat.size, lon.size), dtype=np.float32)
+
+    T[:, 0, :, :] = 25.0 # TODO: make sure the units are correct!
```
Comment (erikvansebille) [inline docs/examples/idealised_flow.py:131]: Has this been done now?

File: docs/examples/idealised_flow.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    S[:, 0, :, :] = 35.0
+
+    for d in range(1, depth.size):
+        T[:, d, :, :] = T[:, 0, :, :] #* ( (depth.size - 0.5 * d) / depth.size)
+        S[:, d, :, :] = S[:, 0, :, :] #* ( (depth.size - 0.5 * d) / depth.size)
```
Comment (erikvansebille) [inline docs/examples/idealised_flow.py:136]: WHy these commented parts of the statements? Are they stray debug messages?

File: docs/examples/idealised_flow.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+    data = {"pp_phyto": pp_phyto, "bio_nanophy": bio_nanophy, "bio_diatom": bio_diatom}
+
+    dimensions = {"lon": lon, "lat": lat, "depth": depth, "time": times}
+    allow_time_extrapolation = True if len(times) == 1 else False
+    fieldsetbgc = parcels.FieldSet.from_data(
+        data, dimensions, mesh="flat", allow_time_extrapolation=allow_time_extrapolation
+    )
+
+    #fieldsetbgc.add_periodic_halo(zonal=True)
```
Comment (erikvansebille) [inline docs/examples/idealised_flow.py:186]: Remove this commented line?

File: docs/examples/idealised_flow.py [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+    fieldset.add_field(fieldsetbgc.pp_phyto)
+    fieldset.add_field(fieldsetbgc.bio_nanophy)
+    fieldset.add_field(fieldsetbgc.bio_diatom)
+
+    return fieldset
+
+
+def add_wind_field(fieldset, times):
+    """ Horizontal wind field with a sinusoidal variation in time."""
+    lon = fieldset.U.grid.lon # Is this right?
```
Comment (erikvansebille) [inline docs/examples/idealised_flow.py:196]: Yes I think so; comment can be removed?

File: docs/examples/idealised_flow.py [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+
+    data = {"Wind_U": wind_U, "Wind_V": wind_V}
+
+    dimensions = {"lon": lon, "lat": lat, "time": times}
+    allow_time_extrapolation = True if len(times) == 1 else False
+    fieldsetwind = parcels.FieldSet.from_data(
+        data, dimensions, mesh="flat", allow_time_extrapolation=allow_time_extrapolation
+    )
+
+    #fieldsetwind.add_periodic_halo(zonal=True)
```
Comment (erikvansebille) [inline docs/examples/idealised_flow.py:221]: Remove commented message?

File: docs/examples/idealised_flow.py [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+
+    data = {"Stokes_U": stokes_U, "Stokes_V": stokes_V, 'wave_Tp': wave_Tp}
+
+    dimensions = {"lon": lon, "lat": lat, "time": times}
+    allow_time_extrapolation = True if len(times) == 1 else False
+    fieldsetStokes = parcels.FieldSet.from_data(
+        data, dimensions, mesh="flat", allow_time_extrapolation=allow_time_extrapolation
+    )
+
+    #fieldsetStokes.add_periodic_halo(zonal=True)
```
Comment (erikvansebille) [inline docs/examples/idealised_flow.py:256]: Remove commented message?

File: docs/examples/idealised_flow.py [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+
+    #fieldsetwind.add_periodic_halo(zonal=True)
+    fieldset.add_field(fieldsetwind.Wind_U)
+    fieldset.add_field(fieldsetwind.Wind_V)
+
+    return fieldset
+
+def add_stokes_field(fieldset, times):
+    """ Horizontal wave field that varies in time."""
+    lon = fieldset.U.grid.lon # Is this right?
```
Comment (erikvansebille) [inline docs/examples/idealised_flow.py:229]: Remove message?

File: docs/examples/idealised_flow.py (hunk 1 above)
Comment (michaeldenes) [reply docs/examples/idealised_flow.py:131]: Yes, will remove

File: docs/examples/idealised_flow.py (hunk 2 above)
Comment (michaeldenes) [reply docs/examples/idealised_flow.py:136]: I had a depth-varying temp/salt field, but I found it too cumbersome to test with, so I decided on a uniform in spacetime field. Will remove

File: docs/examples/idealised_flow.py (hunk 4 above)
Comment (michaeldenes) [reply docs/examples/idealised_flow.py:196]: Thanks!

File: docs/examples/idealised_flow.py (hunk 5 above)
Comment (michaeldenes) [reply docs/examples/idealised_flow.py:221]: Done

File: docs/examples/idealised_flow.py (hunk 6 above)
Comment (michaeldenes) [reply docs/examples/idealised_flow.py:256]: Done

File: docs/examples/idealised_flow.py (hunk 7 above)
Comment (michaeldenes) [reply docs/examples/idealised_flow.py:229]: Done

**Comment by michaeldenes** [comment]
> * [ ]  Before cell 5, perhaps provide links to other examples where the settings dictionary _can_ be loaded?

I've made a link to the examples page, just using 'see other examples'. Or should we flesh this out more?
> * [ ]  In cell 5, mention where the b/c constants come from? Some values seem arbitrary, otherwise?

I've added to the comment that they come from the Kooi et al, Lobelle et al, and Kaandorp et al papers, with dois.
> * [ ]  Perhaps for a later PR, but could the setting of fieldset-constants from the settings-dict not be a method?

I'm not sure I understand this one just yet, but maybe there's some refactoring we can do, yep!
> * [ ]  In cell 10 (and later), how would the user know it's `kernels[-2]` that needs replacing?

I've added a note for cell 10, but have left all the later ones as is. Or should we add the note to each?
> * [ ]  In cell 29, perhaps print the final depth range with fewer digits?

I've made it 4dp now. Perhaps we also print the difference?

**Comment by erikvansebille** [comment]
Thanks for incorporating these, @michaeldenes. Two more comments on the notebook
1. The new example is not listed at https://plasticparcels--55.org.readthedocs.build/en/55/examples.html?
2. The else statement in cell 4 can be removed (instead of commented out)?

**Comment by michaeldenes** [comment]
> Thanks for incorporating these, @michaeldenes. Two more comments on the notebook
> 
> 1. The new example is not listed at https://plasticparcels--55.org.readthedocs.build/en/55/examples.html?
> 2. The else statement in cell 4 can be removed (instead of commented out)?

Thanks @erikvansebille, I've now made these two changes

**Review by erikvansebille — APPROVED** [review]

**Pull request merged**